test/algorithms/qrao/test_quantum_random_access_encoding.py

```diff
@@ -0,0 +1,246 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Tests for QuantumRandomAccessEncoding"""
+import itertools
+import unittest
+from test.optimization_test_case import QiskitOptimizationTestCase
+
+from ddt import ddt, data, unpack
+import numpy as np
+import networkx as nx
+
+from qiskit.circuit import QuantumCircuit
+from qiskit.primitives import Estimator
+from qiskit.quantum_info import SparsePauliOp
+
+from qiskit_optimization.algorithms.qrao import (
+    EncodingCommutationVerifier,
+    QuantumRandomAccessEncoding,
+)
+from qiskit_optimization.problems import QuadraticProgram, QuadraticObjective
+from qiskit_optimization.applications import Maxcut
+
+
+class TestQuantumRandomAccessEncoding(QiskitOptimizationTestCase):
+    """QuantumRandomAccessEncoding tests."""
+
+    def setUp(self):
+        super().setUp()
+        self.problem = QuadraticProgram()
+        self.problem.binary_var("x")
+        self.problem.binary_var("y")
+        self.problem.binary_var("z")
+        self.problem.minimize(linear={"x": 1, "y": 2, "z": 3})
```

Comment:
Could you add a problem instance that contains not only linear terms but also quadratic terms?